code/go/0chain.net/smartcontract/storagesc/handler.go

```diff
@@ -215,17 +204,90 @@ func (ssc *StorageSmartContract) GetBlobbersHandler(
 	}
 
 	var sns storageNodesResponse
-	for _, blobber := range blobbers {
-		sn, err := blobberTableToStorageNode(blobber)
-		if err != nil {
-			return ssc.GetBlobbersHandlerDeprecated(ctx, params, balances)
-		}
-		sns.Nodes = append(sns.Nodes, sn)
+	sns.Nodes = make([]storageNodeResponse, len(blobbers))
+	for i, blobber := range blobbers {
+		sn := blobberTableToStorageNode(blobber)
+		sns.Nodes[i] = sn
 	}
 	return sns, nil
 }
 
-func (msc *StorageSmartContract) GetTransactionByHashHandler(
+// GetAllocationBlobbersHandler returns list of all blobbers alive that match the allocation request.
+func (ssc *StorageSmartContract) GetAllocationBlobbersHandler(
+	ctx context.Context,
+	params url.Values, balances cstate.StateContextI) (interface{}, error) {
+	var err error
+	var creationDate = balances.GetTransaction().CreationDate
+
+	allocData := params.Get("allocation_data")
+	var request newAllocationRequest
+	if err := request.decode([]byte(allocData)); err != nil {
+		return "", common.NewErrInternal("can't decode allocation request", err.Error())
+	}
+
+	var sa = request.storageAllocation()
+	var conf *Config
+	if conf, err = ssc.getConfig(balances, true); err != nil {
+		return nil, fmt.Errorf("can't get config: %v", err)
+	}
+	sa.TimeUnit = conf.TimeUnit // keep the initial time unit
+	if err = sa.validate(creationDate, conf); err != nil {
+		return nil, fmt.Errorf("invalid request: %v", err)
+	}
+	// number of blobbers required
+	var numberOfBlobbers = sa.DataShards + sa.ParityShards
+	// size of allocation for a blobber
+	var allocationSize = sa.bSize()
+	dur := common.ToTime(sa.Expiration).Sub(common.ToTime(creationDate))
+
+	if balances.GetEventDB() == nil {
```

Comment:
Do this event nil checking at the beginning of this function. Perhaps we should have a middleware like stuff that do the event db nil checking before entering the SC functions.